src/Sylius/Component/Grid/Filter/EntityFilter.php

```diff
@@ -29,8 +29,16 @@ public function apply(DataSourceInterface $dataSource, $name, $data, array $opti
         }
 
         $field = isset($options['field']) ? $options['field'] : $name;
+        $fields = isset($options['fields']) ? $options['fields'] : [$field];
 
-        $dataSource->restrict($dataSource->getExpressionBuilder()->equals($field, $data['id']));
+        $expressionBuilder = $dataSource->getExpressionBuilder();
+
+        $expressions = [];
+        foreach ($fields as $field) {
+            $expressions[] = $expressionBuilder->equals($field, $data['id']);;
```

Comment:
If we follow my suggestion from above to make `EntityFilterType` a child of `EntityType`, we'll just need `$data` instead of `$data['id']` here.